samza-kafka/src/main/scala/org/apache/samza/config/KafkaConfig.scala

```diff
@@ -82,11 +87,40 @@ object KafkaConfig {
 
 class KafkaConfig(config: Config) extends ScalaMapConfig(config) {
   // checkpoints
-  def getCheckpointSystem = getOption(KafkaConfig.CHECKPOINT_SYSTEM)
+  def getCheckpointSystem = Option(getOrElse(KafkaConfig.CHECKPOINT_SYSTEM, new JobConfig(config).getDefaultSystem.orNull))
```

Comment:
I noticed this too. Here's what I found. 
Checkpointing was originally designed to be either file-based or stream-based, and there was only 1 factory provided for each. 

If you use the file checkpoint factory, you have to also specify the property
```task.checkpoint.path```

If you use the kafka checkpoint factory, you have to also specify the property
```task.checkpoint.system```

So to support any general System for checkpointing we would need to provide a new factory and move this property and I felt that was outside the scope of this fix.